cpp/include/cuspatial/experimental/detail/haversine.hpp

```diff
@@ -0,0 +1,125 @@
+/*
```

Comment:
Shouldn't this be named `haversine.cuh`?

Reply:
Depends on your style. Thrust doesn't follow that convention.

Reply:
Actually, in the future we may want to enable both host and device execution of these algorithms, a la Thrust.

Reply:
I thought anything with CUDA C++ in it was supposed to either be `.cu` or `.cuh`? If someone includes this from a `.cpp` file, GCC won't know what to do with the `__device__` functions?

Reply:
True. Thrust uses .h for all headers, and makes sure to protect everything so that it can be portable to different backends. I don't know if we want to go that direction. I just have a particular aversion to ".cuh" used in non-detail headers (not sure why, it's just ugly I guess).

Reply:
Generally my rule is that `.h/.hpp` files should be able to included in a host-only TU without issue.

I'd go with `.cuh` unless we decide to make cuSpatial work on both CPU/GPU via Thrust.

Reply:
Done. It would be nice to support CPU and GPU backends but I'm not sure how feasible that is with the more complicated algorithms. Also it would have a large impact on tests, which currently use device vectors. So this is definitely future work.